modules/api/pkg/client/manager.go

```diff
@@ -358,25 +363,38 @@ func (self *clientManager) buildCmdConfig(authInfo *api.AuthInfo, cfg *rest.Conf
 // Extracts authorization information from the request header
 func (self *clientManager) extractAuthInfo(req *restful.Request) (*api.AuthInfo, error) {
 	authHeader := req.HeaderParameter("Authorization")
-	impersonationHeader := req.HeaderParameter("Impersonate-User")
-	jweToken := req.HeaderParameter(JWETokenHeader)
+	impersonationHeader := req.HeaderParameter(ImpersonateUserHeader)
 
 	// Authorization header will be more important than our token
 	token := self.extractTokenFromHeader(authHeader)
-	if len(token) > 0 {
-
+	if len(token) > 0 || len(impersonationHeader) > 0 {
+		if token == "" {
+			// If Impersonate* header(s) provided, but Authorization token is missing,
+			// fallback to in-cluster config by reading the serviceaccount token from the filesystem
+			var dat, _ = os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+			token = string(dat)
```

Comment:
@floreks is there a way to activate Impersonation when no explicit `Authorization: Bearer` token has been included in the request, i.e. "fallback" to using in-cluster config if the reverse proxy has added Impersonation headers only?

Reply:
Note: would also like to cache the result of `os.ReadFile` so as to not read from the filesystem on every request.